src/test/java/com/datalogics/pdf/hsm/samples/MockedHsmTest.java

```diff
@@ -0,0 +1,223 @@
+/*
+ * Copyright 2016 Datalogics Inc.
+ */
+
+package com.datalogics.pdf.hsm.samples;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import com.adobe.pdfjt.core.credentials.PrivateKeyHolder;
+import com.adobe.pdfjt.core.credentials.PrivateKeyHolderFactory;
+import com.adobe.pdfjt.core.credentials.impl.ByteArrayKeyHolder;
+import com.adobe.pdfjt.core.credentials.impl.utils.CertUtils;
+import com.adobe.pdfjt.core.exceptions.PDFIOException;
+import com.adobe.pdfjt.core.exceptions.PDFInvalidDocumentException;
+import com.adobe.pdfjt.core.exceptions.PDFSecurityException;
+import com.adobe.pdfjt.pdf.document.PDFDocument;
+import com.adobe.pdfjt.services.digsig.SignatureFieldInterface;
+import com.adobe.pdfjt.services.digsig.SignatureManager;
+
+import com.datalogics.pdf.hsm.samples.fakes.AbstractHsmManager;
+import com.datalogics.pdf.hsm.samples.fakes.FakeProvider;
+import com.datalogics.pdf.hsm.samples.util.DocumentUtils;
+import com.datalogics.pdf.hsm.samples.util.LogRecordListCollector;
+import com.datalogics.pdf.security.HsmManager;
+
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.security.Key;
+import java.security.NoSuchAlgorithmException;
+import java.security.Security;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import java.security.spec.InvalidKeySpecException;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.logging.Level;
+import java.util.logging.LogRecord;
+import java.util.logging.Logger;
+
+/**
+ * Unit tests simulating a connected HSM device.
+ */
+public class MockedHsmTest extends SampleTest {
```

Comment:
I did rename several helper classes from `Mock*` to `Fake*`, but I didn't think about this issue. It's true that this class has no "@mock"-ed tests.

I struggled with what to name this test (to distinguish it from `UnconnectedHsmTest`: `ConnectedHsmTest` would be misleading since it doesn't connect to a _real_ HSM. `FakeLunaHsmTest` perhaps?